lib/Dialect/FIRRTL/Transforms/CheckCombLoops.cpp

```diff
@@ -25,533 +26,454 @@
-    while (!worklist.empty()) {
-      Node node = worklist.pop_back_val();
-      for (auto &use : node.val.getUses()) {
-        Operation *op = use.getOwner();
-        TypeSwitch<Operation *>(op)
-            .Case<SubfieldOp>([&](SubfieldOp sub) {
-              auto bundleType = sub.getInput().getType();
-              auto fId = bundleType.getFieldID(sub.getFieldIndex());
-              auto res = sub.getResult();
-              handleSub(res, node.fieldId + fId);
-            })
-            .Case<SubindexOp>([&](SubindexOp sub) {
-              auto vecType = sub.getInput().getType().cast<FVectorType>();
-              auto res = sub.getResult();
-              auto fId = vecType.getFieldID(sub.getIndex());
-              handleSub(res, node.fieldId + fId);
-            })
-            .Case<SubaccessOp>([&](SubaccessOp sub) {
-              auto vecType = sub.getInput().getType().cast<FVectorType>();
-              auto res = sub.getResult();
-              // The result of a subaccess can refer to multiple storage
-              // locations corresponding to all the possible indices.
-              for (size_t index = 0; index < vecType.getNumElements(); ++index)
-                handleSub(
-                    res,
-                    node.fieldId + 1 +
-                        index * (vecType.getElementType().getMaxFieldID() + 1));
-            })
-            .Default([&](auto op) {});
+        // child nodes added, continue the DFS
+        if (dfsSize != dfsStack.size())
+          continue;
+        // Node is an SCC root node, pop the visiting stack to remove the nodes
+        // that are no longer active predecessors, that is their sub-tree is
+        // already explored.
+        if (lowLink[dfsVal] == discoveryTime[dfsVal]) {
+          // All the Values reachable from `dfsVal` have been explored, remove
+          // it and its children from the visiting stack.
+          while (visiting.contains(Node(dfsVal, 0))) {
+            auto visitingVal = visitingStack.pop_back_val();
+            forallRefersTo(visitingVal, [&](Node visitingNode) {
+              LLVM_DEBUG(llvm::dbgs() << "\n Erasing visiting : "
+                                      << getFieldName(visitingNode).first);
+              visiting.erase(visitingNode);
+              return success();
+            });
+          }
+        }
+        auto popped = dfsStack.pop_back_val();
+        LLVM_DEBUG(llvm::dbgs()
+                   << "\n dfs popped :" << getFieldName(Node(popped, 0)).first);
       }
     }
-  }
-
-  bool getValueFromStorageLocId(size_t storageLocId, Value &val) {
-    auto iter = storageLocIdToValuesMap.find(storageLocId);
-    if (iter == storageLocIdToValuesMap.end())
-      return false;
-    if (iter->getSecond().empty())
-      return false;
-    val = iter->getSecond().front();
-    return true;
-  }
+    LLVM_DEBUG(
+        for (auto vI
+             : discoveryTime) {
+          LLVM_DEBUG(llvm::dbgs()
```

Comment:
1) what is clang-format doing here?? :rofl: 

2) Only need one level of `LLVM_DEBUG`.

Reply:
`clang-format` can likely be tricked here with `LLVM_DEBUG({ ... })`.

Reply:
Thanks the `{}` worked.